language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/TypeTestExpressionNodeContext.java

```diff
@@ -97,47 +96,40 @@ public TypeTestExpressionNodeContext() {
             //check for intersection as well.
             return completionItems;
         }
-        completionItems = typeReferences.stream().filter(typeRef -> isQualifiedTypeReference(context, typeRef))
-                .map(CommonUtil::getRawType)
-                .map(typeRef -> {
-                    String symbolRef = getQualifiedSymbolReference(typeRef);
-                    return new SymbolCompletionItem(context, typeRef,
-                            TypeCompletionItemBuilder.build(typeRef, symbolRef));
-                }).collect(Collectors.toList());
+        typeReferences.stream()
+                .filter(typeRef -> isQualifiedTypeSymbol(context, typeRef))
+                .forEach(typeRef -> {
+                    String typeName = CommonUtil.getModifiedTypeName(context, typeRef);
+                    if (!typeName.isEmpty()) {
+                        TypeSymbol rawType = CommonUtil.getRawType(typeRef);
+                        completionItems.add(new SymbolCompletionItem(context, rawType,
+                                TypeCompletionItemBuilder.build(rawType, typeName)));
+                    }
+                });
         return completionItems;
     }
 
-    private String getQualifiedSymbolReference(TypeSymbol typeSymbol) {
-        ModuleID moduleID = typeSymbol.getModule().get().id();
-        String moduleName = moduleID.moduleName();
-        String modulePrefix = moduleID.modulePrefix();
-        if (modulePrefix.isEmpty()) {
-            List<String> moduleNameComponents = Arrays.stream(moduleName.split("\\."))
-                    .map(CommonUtil::escapeModuleName)
-                    .collect(Collectors.toList());
-            if (moduleNameComponents.isEmpty()) {
-                modulePrefix = moduleName;
-            } else {
-                modulePrefix = moduleNameComponents.get(moduleNameComponents.size() - 1);
-            }
-        }
-        return modulePrefix + ":" + typeSymbol.getName().get();
-    }
+    private boolean isQualifiedTypeSymbol(BallerinaCompletionContext context,
```

Comment:
IMO `isQualifiedTypeReference` method name should not be changed here.

Reply:
+1